src/modules/git_commit.rs

```diff
@@ -98,6 +98,38 @@ mod tests {
         repo_dir.close()
     }
 
+    #[test]
+    fn test_ceiling_directory() -> io::Result<()> {
+        let repo_dir = fixture_repo(FixtureProvider::Git)?;
+        let level1 = repo_dir.path().join("level1");
+        let level2 = level1.join("level2");
+        std::fs::create_dir(&level1)?;
+        std::fs::create_dir(&level2)?;
+
+        const CEILING_DIRECTORY_VAR: &str = "GIT_CEILING_DIRECTORIES";
+
+        let prev_ceiling = std::env::var_os(CEILING_DIRECTORY_VAR);
+        std::env::set_var(CEILING_DIRECTORY_VAR, level1.as_os_str());
```

Comment:
I'm not really sure how else I should do this, since gitoxide [uses `std::env::var_os`](https://github.com/Byron/gitoxide/blob/78517a616b4b1b6fefc2367057a9b3b786d93750/git-discover/src/upwards/types.rs#L81-L87).